designPatterns-4/src/creationalpattern/FactoryPattern.java

```diff
@@ -0,0 +1,129 @@
+package creationalpattern;
+
+import java.util.Scanner;
+
+public class FactoryPattern {
+    public static void main(String[] args) {
+        Scanner sc = new Scanner(System.in);
+
+        System.out.println("choose you shop\nchicken\nmutton");
+        String choice = sc.nextLine();
+        //input validation
+        try {
+        	if(choice != "mutton" || choice != "chicken") {
+        		throw new WrongInputException();
+        	}
+        }
+        catch(WrongInputException e) {
+        	System.out.println("Enter valid input");
+        }
+        finally {
+            BiriyaniFactory biriyaniFactoryRR = new RRBiriyani();
```

Comment:
`finally` block will be executed irrespective of the exception occurrence. So, as per your implementation, even if User provides 'fried-rice' as input, it'll throw an exception and proceed to create biriyani, which isn't right. You have to throw an exception and should not execute the rest of the logic

Reply:
changed the logic